lib/gc/Transforms/GPU/Pipeline.cpp

```diff
@@ -37,6 +37,7 @@ void populateGPUPipeline(OpPassManager &pm,
 
   pm.addPass(createDecomposeTensorOperation());
   pm.addNestedPass<func::FuncOp>(createGpuTilingAndFusion());
+  pm.addPass(createCanonicalizerPass());
```

Comment:
We should do the 'cleaning' right after the tiling. Otherwise the bufferization pass can produce `memref.cast` ops that can not be lowered by `memref-to-spirv`